src/Assets.php

```diff
@@ -0,0 +1,193 @@
+	}
+
+	/**
+	 * Get the file modified time as a cache buster if we're in dev mode.
+	 *
+	 * @param string $file Local path to the file.
+	 * @return string The cache buster value to use for the given file.
+	 */
+	protected function get_file_version( $file ) {
+		if ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) {
+			$file = trim( $file, '/' );
+			return filemtime( WGPB_ABSPATH . $file );
+		}
+		return WGPB_VERSION;
+	}
+
+	/**
+	 * Registers a script according to `wp_register_script`, additionally loading the translations for the file.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param string $handle    Name of the script. Should be unique.
+	 * @param string $src       Full URL of the script, or path of the script relative to the WordPress root directory.
+	 * @param array  $deps      Optional. An array of registered script handles this script depends on. Default empty array.
+	 * @param bool   $has_i18n  Optional. Whether to add a script translation call to this file. Default 'true'.
+	 */
+	protected function register_script( $handle, $src, $deps = array(), $has_i18n = true ) {
+		$filename = str_replace( plugins_url( '/', WGPB_PLUGIN_FILE ), '', $src );
+		$ver      = $this->get_file_version( $filename );
+		wp_register_script( $handle, $src, $deps, $ver, true );
+		if ( $has_i18n && function_exists( 'wp_set_script_translations' ) ) {
+			wp_set_script_translations( $handle, 'woo-gutenberg-products-block', WGPB_ABSPATH . 'languages' );
+		}
+	}
+
+	/**
+	 * Registers a style according to `wp_register_style`.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param string $handle Name of the stylesheet. Should be unique.
+	 * @param string $src    Full URL of the stylesheet, or path of the stylesheet relative to the WordPress root directory.
+	 * @param array  $deps   Optional. An array of registered stylesheet handles this stylesheet depends on. Default empty array.
+	 * @param string $media  Optional. The media for which this stylesheet has been defined. Default 'all'. Accepts media types like
+	 *                       'all', 'print' and 'screen', or media queries like '(orientation: portrait)' and '(max-width: 640px)'.
+	 */
+	protected function register_style( $handle, $src, $deps = array(), $media = 'all' ) {
+		$filename = str_replace( plugins_url( '/', WGPB_PLUGIN_FILE ), '', $src );
+		$ver      = $this->get_file_version( $filename );
+		wp_register_style( $handle, $src, $deps, $ver, $media );
+	}
+
+	/**
+	 * Output useful globals before printing any script tags.
+	 *
+	 * These are used by @woocommerce/components & the block library to set up defaults
+	 * based on user-controlled settings from WordPress.
+	 *
+	 * @since 2.0.0
+	 */
+	public function print_script_settings() {
+		global $wp_locale;
+		$code           = get_woocommerce_currency();
+		$product_counts = wp_count_posts( 'product' );
+
+		// NOTE: wcSettings is not used directly, it's only for @woocommerce/components
```

Comment:
Wondering if we should add a TODO here to remove this once wc-admin is in core too... @ryelle thoughts?